pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -412,8 +404,8 @@ def phase_str(phi: float) -> str:
     t_max = final_t * 1.05
 
     for ch, axes in ch_axes.items():
-        ch_obj = seq.declared_channels[ch]
         ch_data = data[ch]
+        ch_obj = sampled_seq._ch_objs[ch]
         ch_eom_intervals = data[ch].eom_intervals
         ch_eom_start_buffers = data[ch].eom_start_buffers
         ch_eom_end_buffers = data[ch].eom_end_buffers
```

Comment:
line 416 can you check that `draw_output` is false in the EOM tutorial for `draw_samples` ?

Reply:
Yes, if I use draw_samples for the EOM tutoral, draw_output is False

Reply:
Images are: first is `draw_samples(sampler.sample(seq))` and second is `draw_samples(sampler.sample(seq, modulation = True))`; I'm also passing the parameters passed to the draws in the notebook (ie: draw_phase_shifts)

## Phase shifts and vz:

![image](https://github.com/pasqal-io/Pulser/assets/1060425/ba4f8eae-90b1-41a2-b2de-dd667c47e785)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/d134f99b-33ec-4e5c-b699-72a8b0fef675)

![image](https://github.com/pasqal-io/Pulser/assets/1060425/ab3185d5-a7e9-459a-a9c3-8a8100b4b882)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/8ac2a0c3-457d-45fa-9f54-994fd1ec8013)

![image](https://github.com/pasqal-io/Pulser/assets/1060425/63d19fbb-80c1-49cf-a4f0-8a7bd3c9318d)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/a5aa379b-140e-4d98-a166-5aa50ca24d66)


## Output Modulation
![image](https://github.com/pasqal-io/Pulser/assets/1060425/32d7f9cb-a009-48ec-822c-3bea93706c71)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/a0fcbe76-10ee-4243-b89e-fa8278af7bb2)

## EOM mode (using the isclose with the relative factor I shown before)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/598537cb-d998-48ad-b3c0-ac9c66a5426d)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/fb87e07b-8cec-4f96-8940-f9fa9a9981b8)


## SLM Mask:
![image](https://github.com/pasqal-io/Pulser/assets/1060425/31932c06-c2af-47a6-828d-94dd9c7b59c0)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/59b44d69-7f5d-4dad-8193-a78257f13664)

![image](https://github.com/pasqal-io/Pulser/assets/1060425/4c1da2ca-160d-4dee-9070-dca1da068954)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/d58c374d-2680-48d3-9e21-1b3f971226c7)

![image](https://github.com/pasqal-io/Pulser/assets/1060425/f6e5c62a-b1b4-4192-b505-54511513383a)
![image](https://github.com/pasqal-io/Pulser/assets/1060425/43537dff-5c48-4eba-837b-4474b063a26e)

Reply:
Thanks for having a look at this, looks good to me !